tracer/src/Datadog.Trace/Util/RuntimeId.cs

```diff
@@ -0,0 +1,122 @@
+// <copyright file="RuntimeId.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Util
+{
+    internal static class RuntimeId
+    {
+        private const string NativeLoaderBaseFilename = "Datadog.AutoInstrumentation.NativeLoader";
+
+#if NETFRAMEWORK
+        private const string NativeLoaderLibName = "Datadog.AutoInstrumentation.NativeLoader.dll";
+#else
+        private const string NativeLoaderLibName = "Datadog.AutoInstrumentation.NativeLoader";
+#endif
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(RuntimeId));
+
+        public static string Get()
+        {
+            if (TryGetRuntimeIdFromNative(out var runtimeId))
+            {
+                Log.Information("Runtime id retrieved from native: " + runtimeId);
+                return runtimeId;
+            }
+
+            var guid = Guid.NewGuid().ToString();
+            Log.Information("Unable to get the runtime id from native. Fallback to Guid.NewGuid() : " + guid);
+
+            return guid;
+        }
+
+        [DllImport(NativeLoaderLibName, CallingConvention = CallingConvention.StdCall, EntryPoint = "GetCurrentAppDomainRuntimeId")]
+        private static extern IntPtr GetRuntimeIdFromNative();
+
+        private static bool TryGetRuntimeIdFromNative(out string runtimeId)
+        {
+            if (!IsNativeLoaderPresent())
```

Comment:
BUG: in case of third party dispatcher (ex: Contrast), the filename set in the COR(E_PROFILER_PATH environment variable will not be OUR loader.
I would remove that check and detect that our loader is not loaded thanks to the exception caught during the P/Invoke call.

Reply:
✔️